### PR TITLE
And shenanigans

### PR DESCRIPTION
**Pull request by erg**

### CODE REVIEW

File: core/kernel/kernel-docs.factor [hunk 1]
```diff
@@ -201,9 +201,22 @@ HELP: and
     { $example "USING: kernel prettyprint ;" "\"hi\" 12.0 and ." "12.0" }
 } ;
 
+HELP: and*
+{ $values { "obj1" "a generalized boolean" } { "obj2" "a generalized boolean" } { "obj1/f" "a generalized boolean" } }
+{ $description "If both inputs are true, outputs " { $snippet "obj2" } ". Otherwise outputs " { $link f } "." }
```
Comment (gifti258) [inline core/kernel/kernel-docs.factor:206]: `obj2` should be `obj1`.

File: core/math/integers/integers.factor [hunk 2]
```diff
@@ -205,7 +205,7 @@ M: bignum (log2) bignum-log2 ; inline
     } cond ; inline
 
 : post-scale ( mantissa scale -- n )
-    [ 2/ ] dip ! drop guard bit
```
Comment (jonenst) [inline core/math/integers/integers.factor:208]: search/replace mishap

File: core/sequences/sequences-docs.factor [hunk 3]
```diff
@@ -1992,7 +1992,7 @@ $nl
 $nl
 "The justification for this is that the VM should not crash if a resizable sequence is resized during the execution of an iteration combinator."
 $nl
-"Indeed, iteration combinators are the primary use-case for these words; if the iteration index is already guarded by a loop test which ensures it is within bounds, then additional bounds checks are redundant. For example, see the implementation of " { $link each } "." ;
```
Comment (jonenst) [inline core/sequences/sequences-docs.factor:1995]: search replace mishap

File: extra/decimals/decimals.factor [hunk 4]
```diff
@@ -43,7 +43,7 @@ SYNTAX: DECIMAL: parse-decimal suffix! ;
 
 ERROR: decimal-types-expected d1 d2 ;
 
-: guard-decimals ( obj1 obj2 -- D1 D2 )
```
Comment (jonenst) [inline extra/decimals/decimals.factor:46]: search replace mishap

File: extra/pcre2/ffi/ffi.factor [hunk 5]
```diff
@@ -409,7 +409,7 @@ FUNCTION-ALIAS: pcre2_set_parens_nest_limit
     int pcre2_set_parens_nest_limit_8 ( pcre2_compile_context* ccontext, uint32_t value )
 
 FUNCTION-ALIAS: pcre2_set_compile_recursion_guard
-    int pcre2_set_compile_recursion_guard_8 ( pcre2_compile_context* ccontext, void* guard_function, void* user_data )
```
Comment (jonenst) [inline extra/pcre2/ffi/ffi.factor:412]: search replace mishap

File: extra/modern/html/html.factor [hunk 6]
```diff
@@ -305,3 +321,68 @@ M: comment walk-html call( obj -- ) ;
             ] [ drop ] if
         ] walk-html
     ] { } make [ payload>> ] map ;
+
```
Comment (jonenst) [inline extra/modern/html/html.factor:324]: are all modern.html changes meant for this PR ?

File: core/kernel/kernel.factor [hunk 7]
```diff
@@ -300,6 +304,18 @@ UNION: boolean POSTPONE: t POSTPONE: f ;
 
 : negate ( quot -- quot' ) [ not ] compose ; inline
 
+: 1check ( ..a x quot: ( ..a x -- ..b ? ) -- ..b ? ) keep swap ; inline
```
Comment (jonenst) [inline core/kernel/kernel.factor:307]: why isn't the stack effect
( ..a x quot: ( ..a x -- ..b ? ) -- ..b x ? )
?

File: basis/io/sockets/secure/openssl/openssl.factor [hunk 8]
```diff
@@ -153,14 +156,15 @@ ERROR: file-expected path ;
         [ handle>> ]
         [
             config>>
-            [ ca-file>> dup [ ssl-file-path ] when ]
-            [ ca-path>> dup [ ssl-file-path ] when ] bi
+            [ [ ca-file>> ] 1check [ ssl-file-path ] when ]
```
Comment (jonenst) [inline basis/io/sockets/secure/openssl/openssl.factor:159]: not equivalent when the dup is after the function ?

File: basis/furnace/actions/actions.factor [hunk 9]
```diff
@@ -51,7 +51,7 @@ CONSTANT: revalidate-url-key "__u"
 
 : revalidate-url ( -- url/f )
     revalidate-url-key param
-    dup [ >url ensure-port [ same-host? ] keep and ] when ;
+    [ >url ensure-port [ same-host? ] verify ] ?call ;
```
Comment (jonenst) [inline basis/furnace/actions/actions.factor:54]: so this is a WIP and verify has been renamed? Sorry was juste browsing some PRs without any specific goal

**Comment by mrjbq7** [comment]
Merged the shenANDigans, lets see how we like it.  Needed a couple small fixes, build farm running soon.

**Pull request closed without merging**